recipes/solr-deploy.rb

```diff
@@ -38,7 +38,7 @@
 if not system("grep '^#{account_name}:x:2001' /etc/passwd") then
     bash "Stop Solr processes before modifying account" do
         code <<-EOS
-            ps -U solr -o pid |tail -1 |xargs kill
+            ps -U solr -o pid |tail +1 |xargs kill
             for {1..30}; do
                 ps -U solr > /dev/null 2>&1 || break
```

Comment:
your ps command may still need the tail +1 also so that it removes the table names.
I'd suggest using the same command as the kill line


looking through google and co, this looks nice

https://stackoverflow.com/a/32449276/3217974
```
Use the kill -0 in a loop, for waiting for terminate all processes + do anything between checks:

function anywait_w_status()
{
    for pid in "$@"
    do
        while kill -0 "$pid"
        do
            echo "Process $pid still running..."
            sleep 1
        done
    done
    echo 'All processes terminated'
}
```

Reply:
Line 43 doesn't even use the output lines, so it can't need `tail`. It just checks the exit code.

I have confirmed that `ps` will return 0 if it found results, and 1 if it did not.